src/chapters/aspdotnet-model-classes/exercises.rst

```diff
@@ -0,0 +1,101 @@
+Exercises: Edit Model Classes
+=============================
+
+Add functionality to edit event objects in your ``CodingEvents`` application. 
+These exercises assume that you have added all of the code from this section of the book and your 
+application resembles the `model-binding branch <https://github.com/LaunchCodeEducation/CodingEventsDemo/tree/model-binding>`__.
+
+The edit form will resemble the form used to create an event.
+
+.. admonition:: Tip 
+
+   As you work through these steps, test your code along the way! 
+   With each change you apply to your code, ask yourself what you expect to see when the application
+   is run. You may not find that all of the steps result in observable changes, though.
+   Use Visual Studio’s debugging tools and read your error messages if you run into issues after applying any of
+   the changes.
+
+#. Create the two handler methods listed below in ``EventsController``. We’ll add code
+   to these in a moment, so just put the outline in place for
+   now.
+
+   #. Create a method to display an edit form with this signature:
+
+      .. sourcecode:: csharp
+         :linenos:
+
+         public IActionResult Edit(int eventId = 1) {
+            // controller code will go here
+         }
+
+   #. Create a method to process the form with this signature:
+
+      .. sourcecode:: csharp
+         :linenos:
+
+         public IActionResult EditEvent(int eventId, string name, string desc) {
+            // controller code will go here
+         }
+
+#. Add the necessary annotations to these methods for them to both live
+   at the path ``/Events/Edit``.
+    
+   #. Judging by the names of the handlers, which should handle ``GET`` requests and which should 
+      handle ``POST`` requests?
+
+   #. You’ll need to configure the route for ``Edit()`` to include the path variable ``eventId``, 
```

Comment:
the solution code doesnt use a path variable at this time - im only able to reach an edit view for the default eventId value, 1. the way i understand this feature, the path should only be reachable with a user-provided event id. that value comes from the user selecting which event they intend to edit. In order to select that, the "edit event" button should be tied to an individual event table listing, rather than a view of all events to edit (ie, the way the delete functionality works right now). 
Can you clarify how you want this app to work or if im missing a bit of info? If I'm not missing something, the java solution might be helpful for spotting some differences - or I can clarify later

Reply:
Fixed! The appropriate tag helper passes in the id as a query parameter so clarified that and adding the link as a column.